src/main/java/org/manifold/compiler/back/digital/Netlist.java

```
@@ -0,0 +1,103 @@
+package org.manifold.compiler.back.digital;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.manifold.compiler.ConnectionType;
+import org.manifold.compiler.ConnectionValue;
+import org.manifold.compiler.PortTypeValue;
+import org.manifold.compiler.PortValue;
+import org.manifold.compiler.TypeMismatchException;
+import org.manifold.compiler.UndeclaredIdentifierException;
+import org.manifold.compiler.middle.Schematic;
+
+import com.google.common.collect.ImmutableMap;
+
+public class Netlist {
+  
+  private ConnectionType digitalWireType;
+  private PortTypeValue digitalInType;
+  private PortTypeValue digitalOutType;
+  
+  private Map<String, Net> nets = new HashMap<>();
+  public Map<String, Net> getNets() {
+    return ImmutableMap.copyOf(nets);
+  }
+  private Map<PortValue, Net> connectedNet = new HashMap<>();
+  public Net getConnectedNet(PortValue port) {
+    if (connectedNet.containsKey(port)) {
+      return connectedNet.get(port);
+    } else {
+      // TODO(murphy) throw an exception?
```

Please do. Better to throw the exception and remove it later if it becomes annoying/problematic.